compiler/resolution/implementForallIntents.cpp

```diff
@@ -130,27 +146,42 @@ static void findOuterVars(BlockStmt* block, SymbolMap& uses) {
 // so an ordered traversal of the two lists gives matching pairs.
 // Count the number of these variables into 'numOuterVars'.
 //
-static void createShadowVars(SymbolMap& uses, int& numOuterVars,
+static void createShadowVars(SymbolMap& uses,
+                             int& numShadowVars, int& totOuterVars,
                              std::vector<Symbol*>& outerVars,
                              std::vector<Symbol*>& shadowVars)
 {
-  numOuterVars = 0;
+  numShadowVars = totOuterVars = 0;
 
   form_Map(SymbolMapElem, e, uses) {
     if (e->value != markPruned) {
+      totOuterVars++;
       Symbol* ovar = e->key;
       // If ovar is a reference, e.g. an index variable of
       // a 'var' iterator, we do not want to force
       // that ref type onto 'svar'. Otherwise the generated
       // code will store into *svar without initializing svar
       // first. Todo: what if ovar is a domain?
-      Symbol* svar = new VarSymbol(ovar->name, ovar->type->getValType());
-      svar->addFlag(FLAG_CONST); // todo: replace with arg intents
-      outerVars.push_back(ovar);
-      shadowVars.push_back(svar);
-      INT_ASSERT(e->value == markUnspecified); // meaning e->value is not set, yet
-      e->value = svar;
-      numOuterVars++;
+      Type* valtype = ovar->type->getValType();
+      IntentTag tiIntent = INTENT_BLANK;
+      ArgSymbol* tiMarker = toArgSymbol(e->value);
+      if (tiMarker)
+        tiIntent = tiMarker->intent;
+      else
+        INT_ASSERT(e->value == markUnspecified);
+      tiIntent = concreteIntent(tiIntent, valtype);
+      if (tiIntent & INTENT_FLAG_REF) {
+        // for now, skip [const-]ref-intented variables altogether
+        // todo: add them in
+        e->value = markPruned;
```

Comment:
Just to record your current thinking, you might elaborate on the possible ill effects of leaving const ref variables off the list.  Does the compiler have the freedom to choose const ref or in intent depending on which is optimal?